etna/lib/etna/clients/magma/formatting/models_csv.rb

```diff
@@ -0,0 +1,250 @@
+module Etna
+  module Clients
+    class Magma
+      class ModelsCsv
+        COLUMNS = [
+            :comments,
+            :model_name, :identifier, :parent_model_name, :parent_link_type,
+            :attribute_name,
+            :attribute_type,
+            :link_model_name,
+            :description,
+            :display_name,
+            :format_hint,
+            :restricted,
+            :read_only,
+            :options,
+            :match,
+            :attribute_group,
+            :hidden,
+            :unique,
+        ]
+
+        COLUMN_AS_BOOLEAN = -> (s) { ['true', 't', 'y', 'yes'].include?(s.downcase) }
+
+        COLUMNS_TO_ATTRIBUTES = {
+            attribute_name: :attribute_name,
+            attribute_type: [:attribute_type, -> (s) { AttributeType.new(s) }],
+            link_model_name: :link_model_name,
+            description: :desc,
+            display_name: :display_name,
+            format_hint: :format_hint,
+            restricted: [:restricted, COLUMN_AS_BOOLEAN],
+            read_only: [:read_only, COLUMN_AS_BOOLEAN],
+            options: [:validation, -> (s) { {"type" => "Array", "value" => s.split(',').map(&:chomp)} }],
+            match: [:validation, -> (s) { {"type" => "Regexp", "value" => Regexp.new(s).source} }],
+            attribute_group: :attribute_group,
+            hidden: [:hidden, COLUMN_AS_BOOLEAN],
+            unique: [:unique, COLUMN_AS_BOOLEAN],
+        }
+
+        def self.apply_csv_row(models = Models.new, row = {}, &err_block)
```

Comment:
A reducer function that accumulates csv row changes into the given models.  Forms the backbone of the csv parser.